objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -390,6 +401,302 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
        }
    }
    
+   /**
+    * Uploads requested images to VANTIQ. Called when query parameter "operation" is set to "save".
+    * @param request        The parameters sent with the query.
+    * @param replyAddress   The replyAddress used to send a query response.
+    */
+   public void uploadLocalImages(Map<String, ?> request, String replyAddress) {       
+       Map<String,Object> parsedParameterVals = handleQueryParameters(request, replyAddress, "upload");
+       if (parsedParameterVals == null) {
+           return;
+       }
+       
+       String imageDir = (String) parsedParameterVals.get("imageDir");
+       String imageName = (String) parsedParameterVals.get("imageName");
+       List<String> imageDate = (List) parsedParameterVals.get("imageDate");
+       ImageUtil imageUtil = (ImageUtil) parsedParameterVals.get("imageUtil");
+       
+       // Processing the query with all the extracted parameters
+       processUploadQuery(imageDir, imageName, imageDate, imageUtil, replyAddress);
+   }
+   
+   /**
+    * A helper method called by uploadLocalImages and deleteLocalImages, used to check the query parameters
+    * @param request        The parameters sent with the query.
+    * @param replyAddress   The replyAddress used to send a query response.
+    * @param operation      The operation calling this method, either "upload" or "delete"
+    * @return
+    */
+   public Map<String, Object> handleQueryParameters(Map<String, ?> request, String replyAddress, String operation) {
+       String imageDir = null;
+       String imageName = null;
+       List<String> imageDate = null;
+       ImageUtil imageUtil;
+       // Checking to make sure there is an image directory specified. (Required)
+       if (!(request.get("imageDir") instanceof String)) {
+           client.sendQueryError(replyAddress, "io.vantiq.extsrc.objectRecognition.invalidQueryRequest", 
+                   "No imageDir was specified, or imageDir was incorrectly specified. "
+                   + "Cannot select image(s) to be "+ operation + " .", null);
+           return null;
+       } else {
+           imageDir = (String) request.get("imageDir");
+           // Set up ImageUtil properties
+           imageUtil = setupQueryImageUtil(request);
```

Comment:
This is not where I'd expect this to happen, but that's the way it works.  We can add this to the list of restructuring necessary.